cmd/nodecmd/create.go

```diff
@@ -78,21 +78,26 @@ func getNewKeyPairName(ec2Svc *ec2.EC2) (string, error) {
 
 // createClusterNodeConfig creates node config and save it in .avalanche-cli/nodes/{instanceID}
 // also creates cluster config in .avalanche-cli/nodes storing various key pair and security group info for all clusters
-func createClusterNodeConfig(nodeID, region, ami, keyPairName, certPath, sg, eip, clusterName string) error {
-	nodeConfig := models.NodeConfig{
-		NodeID:        nodeID,
-		Region:        region,
-		AMI:           ami,
-		KeyPair:       keyPairName,
-		CertPath:      certPath,
-		SecurityGroup: sg,
-		ElasticIP:     eip,
-	}
-	err := app.CreateNodeCloudConfigFile(nodeID, &nodeConfig)
-	if err != nil {
-		return err
+func createClusterNodeConfig(nodeIDs, publicIPs []string, region, ami, keyPairName, certPath, sg, clusterName string) error {
+	for i := range nodeIDs {
+		nodeConfig := models.NodeConfig{
+			NodeID:        nodeIDs[i],
+			Region:        region,
+			AMI:           ami,
+			KeyPair:       keyPairName,
+			CertPath:      certPath,
+			SecurityGroup: sg,
+			ElasticIP:     publicIPs[i],
+		}
+		err := app.CreateNodeCloudConfigFile(nodeIDs[i], &nodeConfig)
+		if err != nil {
+			return err
+		}
+		if err = updateClusterConfig(nodeIDs[i], keyPairName, certPath, clusterName); err != nil {
```

Comment:
I believe it is updating keypair and certpath many times, probably we need two cluster config functions, one for adding a key pair and another for adding a node. The problem for the reader is that he may think setting the key/cert is needed in the loop